### PR TITLE
 Blog App: validations, model specs, and n+1 problems

### PR DESCRIPTION
**Pull request by Ntwali-Josue**
### In this pull request, I: 

- [x] Add Validation, specs for all models and fix N+1 queries problems.

### CODE REVIEW

File: config/initializers/simple_form.rb [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  # collection wrappers.
+  # config.include_default_input_wrapper_class = true
+
+  # Defines which i18n scope will be used in Simple Form.
+  # config.i18n_scope = 'simple_form'
+
+  # Defines validation classes to the input_field. By default it's nil.
+  # config.input_field_valid_class = 'is-valid'
+  # config.input_field_error_class = 'is-invalid'
+end
```
Comment (DeliceLydia) [inline config/initializers/simple_form.rb:176]: - [ ] I am not able to create a database because of this file please fix that error
j

Comment (DeliceLydia) [reply config/initializers/simple_form.rb:176]: Please work on the validations
- [ ] User
- Name must not be blank
- PostsCounter must be an integer greater than or equal to zero.

- [ ] Post
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.

- [ ] Work on validations tests
- [ ] Work on flash messages
- [ ] Remember to work on `N+1` problem

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Ntwali-Josue, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Major highlights for your project:
- Kudos to the good README
- Followed git workflow
- No linter errors


## Required Changes ♻️
_Check the comments under the review._




### You Rock!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**